### ISSUE TITLE
1.9.0 chart issue with repo-config in air-gapped env

### ISSUE THREAD

**Issue by yug0slav** [NVIDIA/gpu-operator#299]
- Issue when configuring local repo configMap for air-gapped env using 1.9.0 operator chart. Default CentOS and cuda repos are still being used/configured in /etc/yum.repos.d/ in nvidia-driver-daemonset pod

```
...
Stopping NVIDIA persistence daemon...
Unloading NVIDIA driver kernel modules...
Unmounting NVIDIA driver rootfs...
Checking NVIDIA driver packages...
Updating the package cache...
Repository base is listed more than once in the configuration
Repository updates is listed more than once in the configuration
Repository extras is listed more than once in the configuration
Repository centosplus is listed more than once in the configuration
Repository cuda is listed more than once in the configuration
Could not retrieve mirrorlist http://mirrorlist.centos.org/?release=7&arch=x86_64&repo=os&infra=container error was
14: curl#7 - "Failed to connect to 2001:4178:5:200::10: Network is unreachable"
...
Cannot find a valid baseurl for repo: base/7/x86_64
Stopping NVIDIA persistence daemon...
Unloading NVIDIA driver kernel modules...
Unmounting NVIDIA driver rootfs...
```

**Comment by cdesiniotis** [comment]
Hi @yug0slav 

Can you try naming your repo configuration files `CentOS-Vault.repo` and `cuda.repo` and retry this? Naming them as such will replace the existing repo configuration files in the driver image at `/etc/yum.repos.d/`, which will avoid this error. 

`apt` will raise a warning for repo's it cannot reach, but it will not fail. `yum` appears to behave differently.

**Comment by yug0slav** [comment]
* cert-config CM described in air-gapped docs isn't working...  
  * looks like certs are being dropped into /etc/pki/ca-trust/extracted/pem/ but not actually installed.
  * drop into /etc/pki/ca-trust/sources/anchors/ and update-ca-trust instead???
 
* switched to HTTP and change repo-config to make it work
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: repo-config
  namespace: gpu-operator
data:
  CentOS-Base.repo: |
    [base]
    name=CentOS-7 - Base
    ...

  CentOS-Updates.repo: |
    [updates]
    name=CentOS-7 - Updates
    ...

  CentOS-Extras.repo: |
    [extras]
    name=CentOS-7 - Extras
    ...

  CentOS-Plus.repo: |
    [centosplus]
    name=CentOS-7 - Plus
    ...
  
  EPEL.repo: |
    [epel]
    name=CentOS-7 - EPEL
    ...
  
  cuda.repo: |
    [cuda]
    name=cuda
    ...
```

**Comment by cdesiniotis** [comment]
Thanks for more details. If I am understanding you correctly, there are two issues. Correct me if I am wrong.

1. On CentOS 7, you have to name your repo config files exactly the same as the original ones you are trying to replace (e.g. `CentOS-Base.repo`, `cuda.repo`). `yum` will complain about duplicate repo entries otherwise.
2. Mounting custom keys/certificates through the `cert-config` ConfigMap is not functional.

For 1, we will improve our documentation.

For 2, we will need to bring up a proper "air-gapped" environment for CentOS 7 and get back to you. Are you able to use HTTP for now and successfully install GPU Operator 1.9 in your air-gapped environment?

**Comment by yug0slav** [comment]
I am using HTTP for now... and yes cert-config isn't functioning in centos pod.

**Comment by cdesiniotis** [comment]
@chrisholzheimer 
> apt seem to only raise a warning, but it seems nvidia-driver-daemonset pod still need packages from official repos to work:

Yes, these packages are required for the driver container to work.

> If we rebuild image including our local repositorys the driver pod is working:

Instead of rebuilding the image, you can follow [these instructions](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/appendix.html#local-package-repository) for mounting a custom repo config file into the driver daemonset so that your local repository mirror is used.

**Comment by chrisholzheimer** [comment]
Thank you @cdesiniotis, i did not noticed this in the documentation. That seem to be something really new. However the repo config file works like a charm!

Additionally we need to add our CA trust chain. I tried it like it is described there, but the certificates are not added to the certificate store of the container. I was able to verify our CA files inside of the `/etc/ssl/certs` dir, but they are not added to the file `ca-certificates.crt`. I needed to `cp` the ca files manually to `/usr/local/share/ca-certifcates` and execute `update-ca-certificates` to get it added. For me it seems so, that is not sufficient to just mounting the files into the `/etc/ssl/certs` directory.
